src/Event.cc

```diff
@@ -132,6 +135,10 @@ void EventMgr::QueueEvent(Event* event)
 		}
 
 	++num_events_queued;
+
+	// Only need to fire this if this is the first event in the queue
+	if ( num_events_queued == 1 )
```

Comment:
I actually thought about this again last night (while trying to fall asleep). Because we potentially add new events during Drain() but don't process them, it should probably set a variable at the start of Drain() and then only fire the flare if something new gets added during that.